components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -971,9 +1118,24 @@ mojom::FeedV2Ptr FeedV2Builder::GenerateAllFeed() {
     base::ranges::move(items, std::back_inserter(feed->items));
   };
 
+  std::vector<std::string> eligible_content_groups;
+  for (const auto& channel_id : subscribed_channels) {
+    eligible_content_groups.push_back(channel_id);
+  }
+  // TODO(LorenzoMnto, fallaciousreasoning): What about direct feeds?
+  for (const auto& publisher_id : subscribed_publishers) {
+    eligible_content_groups.push_back(publisher_id);
+  }
+
   // Step 1: Generate a block
   // https://docs.google.com/document/d/1bSVHunwmcHwyQTpa3ab4KRbGbgNQ3ym_GHvONnrBypg/edit#heading=h.rkq699fwps0
-  auto initial_block = GenerateBlock(articles);
+  std::vector<mojom::FeedItemV2Ptr> initial_block;
+  auto initial_content_groups = SampleSubscribedGroups(channels, eligible_content_groups);
```

Comment:
It would be nice if we could hide this `SampleSubscribedGroups` inside `GenerateBlock` - it'd be much easier to use.